src/diagram/components/expression-editor.tsx

```diff
@@ -13,6 +13,14 @@ export const ExpressionEditor: React.FC<IProps> = ({variable, onShowExpressionEd
   const [appliedExpression, setAppliedExpression] = useState(variable.expression);
   const expressionEditorTextAreaRef = useRef<HTMLTextAreaElement | null>(null);
 
+  useEffect(() => {
```

Comment:
Add a comment about why this is necessary and why it's implemented the way it is.